### PR TITLE
Separately add the special user data 'streamOutControlBuf'

### PR DESCRIPTION
**Pull request by amdrexu**
Make a separate path to add this special user data for clearness. And in the future, even if transform feedback is disabled, we still need it for other purposes.

### CODE REVIEW

**Comment by amdvlk-admin** [comment]
### Test summary for commit 9b52eb3b80d7b25ad0ec6f04d6c7b117422478a2
<details><summary>CTS tests (Failed: 172/209067)</summary><ul><li>Built with version 1.3.3.1</li>
<details open><summary>Rhel 8.9, Gfx10</summary><ul>

<li>Passed: 34736/69788 (49.8%)</li>
<li>Failed: 0/69788 (0.0%)</li>
<li>Not Supported: 35052/69788 (50.2%)</li><li>Warnings: 0/69788 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 38448/69504 (55.3%)</li>
<li><details><summary>Failed: 172/69504 (0.2%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.transform_feedback.fuzz.2_level_array.vec4.geometry
Stack trace: Fail

FAILURE: dEQP-VK.transform_feedback.fuzz.2_level_struct_array.instance_array_geometry
Stack trace: Fail

FAILURE: dEQP-VK.transform_feedback.fuzz.3_level_array.dmat2x4.geometry
Stack trace: Fail

FAILURE: dEQP-VK.transform_feedback.fuzz.instance_array_basic_type.dmat2x3.geometry
Stack trace: Fail

FAILURE: dEQP-VK.transform_feedback.fuzz.instance_array_basic_type.dmat2x4.geometry
Stack trace: Fail

FAILURE: dEQP-VK.transform_feedback.fuzz.instance_array_basic_type.dmat3x4.geometry
Stack trace: Fail

FAILURE: dEQP-VK.transform_feedback.fuzz.random_geometry.all_instance_array.16
Stack trace: Fail

FAILURE: dEQP-VK.transform_feedback.fuzz.random_geometry.all_instance_array.36
Stack trace: Fail

FAILURE: dEQP-VK.transform_feedback.fuzz.random_geometry.all_instance_array.41
Stack trace: Fail

FAILURE: dEQP-VK.transform_feedback.fuzz.random_geometry.all_instance_array.57
Stack trace: Fail

FAILURE: dEQP-VK.transform_feedback.fuzz.random_geometry.all_instance_array.68
Stack trace: Fail

FAILURE: dEQP-VK.transform_feedback.fuzz.random_geometry.all_instance_array.72
Stack trace: Fail

FAILURE: dEQP-VK.transform_feedback.fuzz.random_geometry.all_instance_array.74
Stack trace: Fail

FAILURE: dEQP-VK.transform_feedback.fuzz.random_geometry.all_instance_array.92
Stack trace: Fail

FAILURE: dEQP-VK.transform_feedback.fuzz.random_geometry.all_instance_array.98
Stack trace: Fail

FAILURE: dEQP-VK.transform_feedback.fuzz.random_geometry.all_missing.14
Stack trace: Fail

FAILURE: dEQP-VK.transform_feedback.fuzz.random_geometry.all_missing.18
Stack trace: Fail

FAILURE: dEQP-VK.transform_feedback.fuzz.random_geometry.all_missing.2
Stack trace: Fail

FAILURE: dEQP-VK.transform_feedback.fuzz.random_geometry.all_missing.28
Stack trace: Fail

FAILURE: dEQP-VK.transform_feedback.fuzz.random_geometry.all_missing.49
Stack trace: Fail
...
```
</p></details></li>

<li>Not Supported: 30884/69504 (44.4%)</li><li>Warnings: 0/69504 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 38654/69775 (55.4%)</li>
<li>Failed: 0/69775 (0.0%)</li>
<li>Not Supported: 31121/69775 (44.6%)</li><li>Warnings: 0/69775 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by amdvlk-admin** [comment]
### Test summary for commit b3188af12d5fd50cf8d84687dd49ba6b28ab15b4
<details><summary>CTS tests (Failed: 0/209351)</summary><ul><li>Built with version 1.3.3.1</li>
<details open><summary>Rhel 9.0, Gfx10</summary><ul>

<li>Passed: 34735/69788 (49.8%)</li>
<li>Failed: 0/69788 (0.0%)</li>
<li>Not Supported: 35053/69788 (50.2%)</li><li>Warnings: 0/69788 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 38620/69788 (55.3%)</li>
<li>Failed: 0/69788 (0.0%)</li>
<li>Not Supported: 31168/69788 (44.7%)</li><li>Warnings: 0/69788 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 38654/69775 (55.4%)</li>
<li>Failed: 0/69775 (0.0%)</li>
<li>Not Supported: 31121/69775 (44.6%)</li><li>Warnings: 0/69775 (0.0%)</li>
</ul></details>


</ul></details>

File: lgc/patch/PatchEntryPointMutate.cpp [hunk 1]
```diff
@@ -1549,45 +1549,53 @@ void PatchEntryPointMutate::addSpecialUserDataArgs(SmallVectorImpl<UserDataArg>
...
+    case ShaderStageTessEval:
+      specialUserDataArgs.push_back(UserDataArg(builder.getInt32Ty(), "streamOutControlBuf",
+                                                UserDataMapping::StreamOutControlBuf,
+                                                &intfData->entryArgIdxs.tes.streamOutData.controlBufPtr));
+      break;
+    case ShaderStageGeometry:
+      specialUserDataArgs.push_back(UserDataArg(builder.getInt32Ty(), "streamOutControlBuf",
+                                                UserDataMapping::StreamOutControlBuf,
+                                                &intfData->entryArgIdxs.gs.streamOutData.controlBufPtr));
+      break;
```
Comment (nhaehnle) [inline lgc/patch/PatchEntryPointMutate.cpp:1593]: Can this be factored further? The only difference here is which field of entryArgIdxs is written, everything else is the same.

Comment (amdrexu) [reply lgc/patch/PatchEntryPointMutate.cpp:1593]: Sure. Will do.

**Comment by amdvlk-admin** [comment]
### Test summary for commit 818b45a7c6ee5e82a2410e7a643098bac817c35e
<details><summary>CTS tests (Failed: 0/209351)</summary><ul><li>Built with version 1.3.3.1</li>
<details open><summary>Rhel 9.0, Gfx10</summary><ul>

<li>Passed: 34735/69788 (49.8%)</li>
<li>Failed: 0/69788 (0.0%)</li>
<li>Not Supported: 35053/69788 (50.2%)</li><li>Warnings: 0/69788 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 38620/69788 (55.3%)</li>
<li>Failed: 0/69788 (0.0%)</li>
<li>Not Supported: 31168/69788 (44.7%)</li><li>Warnings: 0/69788 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 38654/69775 (55.4%)</li>
<li>Failed: 0/69775 (0.0%)</li>
<li>Not Supported: 31121/69775 (44.6%)</li><li>Warnings: 0/69775 (0.0%)</li>
</ul></details>


</ul></details>

**Review by nhaehnle — APPROVED** [review]
LGTM, thanks!

**Pull request merged**